.github/workflows/main.yml

```diff
@@ -82,7 +82,6 @@ jobs:
       - name: Set up repos
         run: |
           set -uexo pipefail
-          ln -s "$GITHUB_WORKSPACE" ../dmd
```

Comment:
@ibuclaw: This seems to fix the coverage issue indeed. In random https://github.com/dlang/dmd/actions/runs/6909318180/job/18800406777, the upload-coverage-report step is ~23k lines long, whereas in https://github.com/dlang/dmd/actions/runs/6915011820/job/18813482794?pr=15831, it's a mere 650 lines.